src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -1,110 +1,435 @@
+            # continue as with non per channel sensor
+            port_name = if_alias
+    else:
+        port_name = if_alias
+
+    return "DOM {} Sensor for {}".format(SENSOR_NAME_MAP[sensor], port_name)
 
 
 class PhysicalTableMIBUpdater(MIBUpdater):
+    """
+    Updater class for physical table MIB
+    """
 
-    DEVICE_METADATA = "DEVICE_METADATA|localhost"
     CHASSIS_ID = 1
+    TRANSCEIVER_KEY_PATTERN = mibs.transceiver_info_table("*")
 
     def __init__(self):
         super().__init__()
 
         self.statedb = SonicV2Connector()
         self.statedb.connect(self.statedb.STATE_DB)
 
-        # List of available sub IDs.
-        self.physical_classes = []
+        self.if_alias_map = {}
+
+        # List of available sub OIDs.
+        self.physical_entities = []
+
         # Map sub ID to its data.
         self.physical_classes_map = {}
+        self.physical_description_map = {}
+        self.physical_hw_version_map = {}
+        self.physical_serial_number_map = {}
+        self.physical_mfg_name_map = {}
+        self.physical_model_name_map = {}
+
+        self.pubsub = None
 
     def reinit_data(self):
         """
         Re-initialize all data.
         """
-        device_metadata = self.statedb.get_all(self.statedb.STATE_DB, self.DEVICE_METADATA)
 
-        # TODO: Add support for unicode in swsssdk
-        chassis_serial_number = ""
+        # reinit cache
+        self.physical_classes_map = {}
+        self.physical_description_map = {}
+        self.physical_hw_version_map = {}
+        self.physical_serial_number_map = {}
+        self.physical_mfg_name_map = {}
+        self.physical_model_name_map = {}
+
+        # update interface maps
+        _, self.if_alias_map, _, _, _ = \
+            mibs.init_sync_d_interface_tables(SonicV2Connector())
+
+        device_metadata = mibs.get_device_metadata(self.statedb)
+        chassis_sub_id = (self.CHASSIS_ID, )
+        self.physical_entities = [chassis_sub_id]
+
         if not device_metadata or not device_metadata.get(b"chassis_serial_number"):
             chassis_serial_number = ""
         else:
             chassis_serial_number = device_metadata[b"chassis_serial_number"]
-            self.physical_classes = [(self.CHASSIS_ID, )]
-            self.physical_classes_map = {
-                    (self.CHASSIS_ID, ): (PhysicalClass.CHASSIS, chassis_serial_number.decode("utf-8"))
-                }
+
+        self.physical_classes_map[chassis_sub_id] = PhysicalClass.CHASSIS
+        self.physical_serial_number_map[chassis_sub_id] = chassis_serial_number
+
+        # retrieve the initial list of transceivers that are present in the system
+        transceiver_info = self.statedb.keys(self.statedb.STATE_DB, self.TRANSCEIVER_KEY_PATTERN)
+        if transceiver_info:
+            self.transceiver_entries = [entry.decode() \
+                for entry in transceiver_info]
+        else:
+            self.transceiver_entries = []
```

Comment:
I'd suggest to write it as
```
self.transceiver_entries = [entry.decode() for entry in transceiver_info]
```

you don't need if else here

Reply:
transceiver_info could be None so I can get exception "NoneType object is not iterable".
So maybe write `if transceiver_info is not None` so it will be clear why we need that if block?